core/src/main/scala/org/scalamock/function/FakeFunction.scala

```diff
@@ -37,9 +37,14 @@ abstract class FakeFunction(protected val mockContext: MockContext, private[scal
   private def expectationContext = mockContext.expectationContext
 
   def handle(arguments: Product): Any = {
-    val call = new Call(this, arguments)
-    callLog += call
-    expectationContext.handle(call) getOrElse onUnexpected(call)
+    if (Option(callLog).isDefined) {
```

Comment:
OK, changed to `!= null`. 

I would love to see the `null`s removed in favour of something more idiomatic, but not sure I want to tackle that at the moment.